client/miner.c

```diff
@@ -278,7 +397,33 @@ void *miner_net_thread(void *arg)
 					ndata = 0;
 					maxndata = sizeof(struct xdag_field);
 				} else {
-					maxndata = 2 * sizeof(struct xdag_field);
+
+					int has = 0;
+
+					if(g_multi_address) {
+						pthread_mutex_lock(&g_mutex_our_block_list);
+						struct block_list_element * elem = NULL;
+						LL_FOREACH(g_our_blocks_list, elem)
+						{
+							if(memcmp(last->data, elem->hash, sizeof(xdag_hashlow_t)) == 0) {
```

Comment:
That part of code is duplicated. I think we can move it to separate function that returns 1 if memcmp == 0 and 0 in other case.